node.go

```diff
@@ -37,70 +37,157 @@ type Node struct {
 	// Edges should be stored in-order for iteration.
 	// We avoid a fully materialized slice to save memory,
 	// since in most cases we expect to be sparse
-	edges edges
+	bitmap   [4]uint64
+	children []*Node
 }
 
 func (n *Node) isLeaf() bool {
 	return n.leaf != nil
 }
 
-func (n *Node) addEdge(e edge) {
-	num := len(n.edges)
-	idx := sort.Search(num, func(i int) bool {
-		return n.edges[i].label >= e.label
-	})
-	n.edges = append(n.edges, e)
-	if idx != num {
-		copy(n.edges[idx+1:], n.edges[idx:num])
-		n.edges[idx] = e
+// setBit sets the bit for a given label
+func setBit(bitmap *[4]uint64, label byte) {
+	block := label / 64
```

Comment:
You can avoid the division by doing a bit shift, e.g. `label >> 6` will be the same as `label / 64` but should be faster.

Reply:
Thanks 👍

Reply:
Also `% 64` can be replaced by `& 63` which is faster as well.